src/Generator/Passes/CleanCommentsPass.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using CppSharp.AST;
+using CppSharp.Generators.CSharp;
+using System.Text.RegularExpressions;
+
+namespace CppSharp.Passes
+{
+    public class CleanCommentsPass : TranslationUnitPass,ICommentVisitor<bool>
+    {
+        public bool VisitBlockCommand(BlockCommandComment comment)
+        {
+            return true;
+        }
+
+        public override bool VisitDeclaration(Declaration decl)
+        {
+            if (!base.VisitDeclaration(decl))
+                return false;
+
+            if (decl.Comment != null)
+            {
+                var fullComment = decl.Comment.FullComment;
+                VisitFull(fullComment);
+                
+            }
+            return true;
+        }
+
+        public bool VisitFull(FullComment comment)
+        {
+            foreach (var block in comment.Blocks)
+                block.Visit(this);
+
+            return true;
+        }
+        #region Comments Visit
+        public bool VisitHTMLEndTag(HTMLEndTagComment comment)
+        {
+            return true;
+        }
+
+        public bool VisitHTMLStartTag(HTMLStartTagComment comment)
+        {
+            return true;
+        }
+
+        public bool VisitInlineCommand(InlineCommandComment comment)
+        {
+            return true;
+        }
+
+        public bool VisitParagraphCommand(ParagraphComment comment)
+        {
+            Regex regex = new Regex(@"^(<|</)[a-zA-Z][\w\-]*?>?$");
```

Comment:
You have already declared an accessible regex but you still have a copy here, why? The whole point was not to have a repetition.